### DISCUSSION TITLE
How to show only the repos that `type=source` on the Top Languages Card

### DISCUSSION THREAD

**Discussion by yhm-amber** [anuraghazra/github-readme-stats#2614] [Q&A] [answered]
Or how to only show **my** own commits in all repos ?

**Comment by qwerty541** [comment, marked as answer]
If I understand your question correctly, you want the Top Languages Card to show data only for the repositories you created, not for forked. I think that's how it works right now. I tested my guess in the following way: using the [exclude](https://github.com/anuraghazra/github-readme-stats#exclude-individual-repositories) option, I excluded out several of my forked repositories and checked if the data on the card has changed. Everything remains unchanged.

**Reply by qwerty541** [reply]
UPD: i checked code of top languages card data fetcher, it fetches data only about owned repositories. Check it: 

https://github.com/anuraghazra/github-readme-stats/blob/master/src/fetchers/top-languages-fetcher.js#L24

**Reply by pchchv** [reply]
If I understand the @yhm-amber correctly, it is not just about his repositories, but about **his commits** in his repositories. 
I don't know how this works by default, but I suspect that it may simply take data about repositories without considering the **authors of specific commits.**

**Reply by qwerty541** [reply]
About the second question of @yhm-amber, currently total commits count shown in stats card consists of user total commits in public repositories and user restricted contributions count if you set `count_private` option to `true`.

See: https://github.com/anuraghazra/github-readme-stats/blob/master/src/fetchers/stats-fetcher.js

As far as I know, it is technically impossible to divide user restricted contributions into categories.

It means that you can don't use `count_private` option to show only your own commits in public repositories.

**Reply by rickstaa** [reply]
@qwerty541 and @pchchv thanks for answering this discussion! This project is nothing without people like you 💙. To add to your answers; Due to limitations in the GraphQL API, we currently fetch the language stats on a repository and not commit basis (see https://github.com/anuraghazra/github-readme-stats/issues/1801 for more information). As a result also commits of other people in your repositories are included in your language card. I created a feature request at https://github.com/orgs/community/discussions/18230 to improve this behaviour. It will be somewhat improved when the GitHub action is released (see https://github.com/anuraghazra/github-readme-stats/issues/2179).

**Reply by qwerty541** [reply]
@yhm-amber Please mark this discussion as answered if you don't need help anymore.
